terraform/spec/dependabot/terraform_spec.rb

```diff
@@ -51,5 +51,19 @@
 
       it { is_expected.to eq("distribution_label::terraform-null-label") }
     end
+
+    context "git unknown source with ref" do
+      let(:name) do
+        "module_name::unknown_provider::repo_name/unknown_name(9685A3B07E8D9C45BE0A3D92B02F13978FB311D8)::tags/0.1.0"
+      end
+
+      it { is_expected.to eq("module_name::unknown_name(9685A3B07E8D9C45BE0A3D92B02F13978FB311D8)") }
+    end
+
+    context "git unknown source without ref" do
+      let(:name) { "module_name::unknown_provider::repo_name/unknown_name(9685A3B07E8D9C45BE0A3D92B02F13978FB311D8)" }
+
+      it { is_expected.to eq("module_name::unknown_name(9685A3B07E8D9C45BE0A3D92B02F13978FB311D8)") }
+    end
```

Comment:
These `display names` are shown in PR titles and might be too long if we include the SHA. 
If we do not know the source maybe we should just use the `module_name`? What do you think?

This would require further modifying the `register_display_name_builder` in https://github.com/dependabot/dependabot-core/blob/main/terraform/lib/dependabot/terraform.rb

Reply:
I can definitely strip the hash, I just assumed it would truncate it if to long.  It doesn't provide any other function other than uniqueness and I believe PRs can have duplicate titles, can you confirm that is true @Nishnha

Reply:
Yep, PRs can have duplicate titles, and Dependabot uses a unique id (not the title) to interact with PRs anyway.

Removing the hash seems like a good move regardless since we should avoid having an arbitrary SHA in the title of a Dependabot PR. Descriptive is better 😃